lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -766,14 +766,18 @@ void ProxyRenderDelegate::_Execute(const MHWRender::MFrameContext& frameContext)
 #endif
 #endif // defined(MAYA_ENABLE_UPDATE_FOR_SELECTION)
 
+    // Work around for USD issue #1516. We don't know if any instanced object has
+    // had it's instance index change, so re-populate selection every update to
+    // ensure correct selection highlighting of instanced objects.
+    bool instanceIndexChanged = true;
```

Comment:
This change fixes the incorrect selection highlight display. Unfortunately there is not a simple way to know if the Instance Index of a selected rprim has changed, so we're forced to call populateSelection every time.

The performance of calling populateSelection every time is not terrible, but it is also not great. The operation is linear in the number of entries in the selection list. When adding/removing rprims on the selection list the populateSelection time is a tiny fraction of the overall time. When the selection list is not changing and we're now forced to call populateSelect the performance cost is real but small. We can't create really long selection lists right now due to other scalability problems, so I think the best approach is to leave it populating selection every frame and then re-visit when we re-examine long selection list performance as a part of MAYA-114875.